drift-client/src/main/java/io/airlift/drift/client/DriftMethodInvocation.java

```diff
@@ -143,7 +144,7 @@ private synchronized void nextAttempt()
 
             Optional<A> address = addressSelector.selectAddress(addressSelectionContext);
             if (!address.isPresent()) {
-                fail();
+                fail("No hosts available");
```

Comment:
Should we change the selector interface to return an address or throw? That would allow useful error messages from the selector. It seems strange to return `Optional` if that is always a failure condition.